### PR TITLE
kitti 3d object detection example for dataset

### PR DESCRIPTION
**Pull request by ly-kolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: examples/dataset/object_detection_3d/object_detection_3d/upload_results.py
Comment (ly-kolena) [inline examples/dataset/object_detection_3d/object_detection_3d/upload_results.py]: This does not display nicely with current UI support, as it is a dictionary. However, included for 2 reasons
1. for direct CSV import, we currently don't have capability to read eval_config from json.
2. for SDK upload, nested object does not expand in detail viewer, and only on "precise hover" over model/eval drop down.

**Comment by ly-kolena** [comment]
Locally with a new conda env, I can run `poetry install` successfully and then run `pytest` with expected output (1 pass + 1 skip) on both linux and osx machine. The poetry resolve process takes a long time, on the order of double digit minutes. Have not successfully get circleci job to pass `poetry install` step. Will follow that up as a separate PR.

File: examples/dataset/object_detection_3d/object_detection_3d/upload_dataset.py
Comment (munkyshi) [inline examples/dataset/object_detection_3d/object_detection_3d/upload_dataset.py]: If we pin our `kolena` version to the latest release, we can skip the `initialize` call

(based on https://github.com/kolenaIO/kolena/pull/537)

**Review by munkyshi — APPROVED** [review]
Approving so this isn't blocked on a Kolena Owner -- @bagalaster will do a more rigorous review on the logic, so I can craft an example and send it out tonight

File: examples/dataset/object_detection_3d/object_detection_3d/upload_results.py
Comment (bagalaster) [inline examples/dataset/object_detection_3d/object_detection_3d/upload_results.py]: Can we reference `VALID_LABELS` instead of hard coding?

File: examples/dataset/object_detection_3d/object_detection_3d/upload_results.py
Comment (bagalaster) [inline examples/dataset/object_detection_3d/object_detection_3d/upload_results.py]: Same here, should this be a top-level constant?

File: examples/dataset/object_detection_3d/object_detection_3d/upload_results.py
Comment (bagalaster) [inline examples/dataset/object_detection_3d/object_detection_3d/upload_results.py]: We could `zip` lists together here instead of indexing into them using the output of `enumerate` - marginally easier to read imho

e.g. `[inference for inference, fp in zip(record.raw_inferences_2d, FP) if fp]`

**Review by bagalaster — COMMENTED** [review]
Just a few nits. I'm pressing the "I believe" button on the hard-core mathematical guts (I imagine we're pretty confident in these just by looking at the data), but the rest of the logic looks good to me.

**Pull request merged**
